### PR TITLE
Adds the Bishop of Sin

### PR DESCRIPTION
**Pull request by sawu-tg**
- Adds a Bishop of Sin, an NPC that when summoned by an Admin, will teleport around the station, randomly judging people for their sins.

Credit to @RemieRichards for the sprite.

**Greed**
Good: Gives 10 assorted "rare" minerals, ie gold, diamond and silver
Bad: Turns the sinner into a statue.
**Gluttony**
Good: Gives 10 assorted snack-food items
Bad: Adds 1000 Nutriment to the sinner, fattening them up.
**Pride**
Good: Gives the Sinner a Hammer of Pride, that transfers all of their reagents to a target when wielded.
Bad: Gives the Sinner a Hammer of Pride, that transfers all of the target's reagents to them when wielded.
**Lust**
Good: Gives a stone that draws everyone and everything nearby towards the sinner.
Bad: Gives a stone that causes a heart attack.
**Envy**
Good: Randomly assigns the sinner the name/appearance of a crewmember
Bad: Randomly assigns the sinner a species
**Sloth**
Good: Puts the sinner to sleep for a short time
Bad: Doses the sinner with Frost oil mimicking the effects of tiredness.
**Wrath**
Good: Doses the sinner up with bath salts to improve their combat efficiency for a short time.
Bad: Doses the sinner up with lexorin and mindbreaker, simulating a psychotic break.

### CODE REVIEW

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+					src.say("Your sin, [sinPerson], is Gluttony.")
+					if(prob(50))
+						src.say("I will indulge your sin, [sinPerson].")
+						var/list/allTypes = list()
+						for(var/A in typesof(/obj/item/weapon/reagent_containers/food/snacks))
+							var/obj/item/weapon/reagent_containers/food/snacks/O = A
+							if(initial(O.cooked_type))
+								allTypes += A
+						for(var/i = 0; i < 10; ++i)
+							var/greed_type = pick(allTypes)
```
Comment (Iamgoofball) [inline code/game/gamemodes/miniantags/hades/hades.dm]: pick_n_take()

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	"The only way to be free is to be free of yourself.",\
+	"Wallow in sin, and give yourself unto darkness.",\
+	"Only the truly sinful may stand.",\
+	"Find yourself and you will find Absolution.",\
+	"Forego the pain of this process, and submit.")
+
+/mob/living/simple_animal/hades/New()
+	..()
+	lastsinPerson = world.time
+	var/list/possible_titles = list("Pope","Bishop","Lord","Cardinal")
```
Comment (RemieRichards) [inline code/game/gamemodes/miniantags/hades/hades.dm]: Deacon.

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+	name = chosen
+	real_name = chosen
+
+	world << "<span class='warning'><font size=6>A [name] has entered your reality. Kneel before them.</font></span>"
+	world << 'sound/effects/pope_entry.ogg'
+
+	Appear()
+
+/mob/living/simple_animal/hades/hitby(atom/movable/AM, skipcatch, hitpush, blocked)
+	..(AM,skipcatch,hitpush,blocked)
```
Comment (RemieRichards) [inline code/game/gamemodes/miniantags/hades/hades.dm]: if you're not altering the arguments, just ```..()``` works fine, automatically passing on all args.

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+						'sound/hallucinations/turn_around1.ogg', 'sound/hallucinations/turn_around2.ogg', 'sound/hallucinations/veryfar_noise.ogg', 'sound/hallucinations/wail.ogg')
+			world << pick(creepyasssounds)
+		else
+			lastsinPerson = world.time
+			var/mob/living/carbon/human/sinPerson = pick(player_list)
+			if(sinPerson)
+				loc = get_turf(pick(oview(1,sinPerson)))
+				Appear()
+				var/sinPersonchoice = pick(list("Greed","Gluttony","Pride","Lust","Envy","Sloth","Wrath"))
+				if(sinPersonchoice == "Greed")
```
Comment (RemieRichards) [inline code/game/gamemodes/miniantags/hades/hades.dm]: switch, please.

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+				else if(sinPersonchoice == "Gluttony")
+					src.say("Your sin, [sinPerson], is Gluttony.")
+					if(prob(50))
+						src.say("I will indulge your sin, [sinPerson].")
+						var/list/allTypes = list()
+						for(var/A in typesof(/obj/item/weapon/reagent_containers/food/snacks))
+							var/obj/item/weapon/reagent_containers/food/snacks/O = A
+							if(initial(O.cooked_type))
+								allTypes += A
+						for(var/i = 0; i < 10; ++i)
```
Comment (RemieRichards) [inline code/game/gamemodes/miniantags/hades/hades.dm]: for(var/i in 1 to 10)

File: code/game/gamemodes/miniantags/hades/hades.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/gamemodes/miniantags/hades/hades.dm]: no? cos that will prevent a greed type from happening, leading to eventually 0 greed types

Comment (Iamgoofball) [inline code/game/gamemodes/miniantags/hades/hades.dm]: its so you dont get the same food twice? it only picks 10 food anyways

Comment (RemieRichards) [inline code/game/gamemodes/miniantags/hades/hades.dm]: well while it won't lead to 0 (I read wrong, thought it was global), what's wrong with a rare chance to get 10 of the same food?

Comment (Iamgoofball) [inline code/game/gamemodes/miniantags/hades/hades.dm]: memes

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+							var/greed_type = pick(allTypes)
+							new greed_type(get_turf(sinPerson))
+					else
+						src.say("Your sin will be punished, [sinPerson]!")
+						sinPerson.reagents.add_reagent("nutriment",9999)
```
Comment (Xhuis) [inline code/game/gamemodes/miniantags/hades/hades.dm]: Max reagents in someone is 1,000, so this probably isn't necessary

**Comment by Niknakflak** [comment]
Define judging

**Comment by Iamgoofball** [comment]
santa

**Comment by sawu-tg** [comment]
@Niknakflak @Iamgoofball 
Pretty much that, he'll teleport around and randomly choose a sin for people, each sin does a different action and has a 50/50 chance of being a bad or good judgement.

They're as follows: 
**Greed**
- Good: Gives 10 assorted "rare" minerals, ie gold, diamond and silver
- Bad: Turns the sinner into a statue.
**Gluttony**
 - Good: Gives 10 assorted snack-food items
 - Bad: Adds 1000 Nutriment to the sinner, fattening them up.
**Pride**
 - Good: Gives station members an objective to praise the sinner
 - Bad: Gives station members an objective to boss the sinner around
**Lust**
 - Good: Gives station members an objective to objectify and adore the sinner
 - Bad: Gives station members an objective to make the sinner like them at any cost
**Envy**
 - Good: Randomly assigns the sinner the name/appearance of a crewmember
 - Bad: Randomly assigns the sinner a species
**Sloth**
 - Good: Puts the sinner to sleep for a short time
 - Bad: Doses the sinner with Frost oil mimicking the effects of tiredness.
**Wrath**
 - Good: Gives the sinner an objective to murder everything
 - Bad: Doses the sinner up with lexorin and mindbreaker, simulating a psychotic break.

**Comment by coiax** [comment]
Oh, I thought it was just going to be a teleporting NPC that just emotes "is judging you." at people.

**Comment by Zilenan91** [comment]
Does this kill people? If it does the attack message should be, "Bishop of Sin learns <x>"

**Comment by sawu-tg** [comment]
@Zilenan91 It can indirectly kill if a judgement is not tended to, but it should never actually directly attack people.

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+							src.say("Your sin will be punished, [sinPerson]!")
+							sinPerson.reagents.add_reagent("nutriment",1000)
+					if("Pride")
+						src.say("Your sin, [sinPerson], is Pride.")
+						if(prob(50))
+							src.say("I will indulge your sin, [sinPerson].")
+							for(var/mob/living/carbon/human/H in player_list)
+								if(H == sinPerson)
+									continue
+								ticker.mode.traitors += H.mind
```
Comment (coiax) [inline code/game/gamemodes/miniantags/hades/hades.dm]: Please no touch gamemode code, there's a freeze on

Comment (sawu-tg) [inline code/game/gamemodes/miniantags/hades/hades.dm]: Is this considered part of a subsystem? I didn't think much would change in terms of the way antagonists work.

**Comment by sawu-tg** [comment]
Closed due to conflict with the freeze.

**Comment by sawu-tg** [comment]
Reliance on traitor system has been removed, re-opening.

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+					user << "You feel insecure, taking on [H]'s burden."
+
+/obj/item/lovestone
+	name = "Stone of Lust"
+	desc = "It lays within your hand, radiating pulses of uncomfortable warmth."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "lovestone"
+	item_state = "lovestone"
+	w_class = 1
+	var/lust_direction = 0
```
Comment (KazeEspada) [inline code/game/gamemodes/miniantags/hades/hades.dm]: This var name is balls. Also, it uses 0 and 1 to determine its behaviour despite being undescriptive as fuck.

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+	desc = "It resonates an aura of Pride."
+	force = 5
+	throwforce = 15
+	w_class = 4
+	slot_flags = SLOT_BACK
+	force_unwielded = 8
+	force_wielded = 18
+	attack_verb = list("attacked", "smashed", "crushed", "splattered", "cracked")
+	hitsound = 'sound/weapons/blade1.ogg'
+	var/pride_direction = 0 // 0 owner > target; 1 target > owner
```
Comment (KazeEspada) [inline code/game/gamemodes/miniantags/hades/hades.dm]: Also bad.

**Comment by Dorsisdwarf** [comment]
why is it called hades

**Comment by sawu-tg** [comment]
@Dorsisdwarf Because Hades, the Greek God, is the Guardian of Death and the Underworld, as well as  the carrier of souls.

The Bishop of Sin is supposed to be a representation of general hell and underworld nonsense, and thus I used the internal name hades.

File: code/game/gamemodes/miniantags/hades/hades.dm (hunk 8 above)
Comment (sawu-tg) [inline code/game/gamemodes/miniantags/hades/hades.dm]: I don't see how this is undescriptive?

The var is x_direction, and it determines the direction that the special effect goes in.

File: code/game/gamemodes/miniantags/hades/hades.dm (hunk 9 above)
Comment (sawu-tg) [inline code/game/gamemodes/miniantags/hades/hades.dm]: ""

**Comment by Iamgoofball** [comment]
@Dorsisdwarf its what I call your mom in bed

File: code/game/gamemodes/miniantags/hades/hades.dm [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+									spawn(10)
+										sinPerson.name = H.name
+										sinPerson.real_name = H.real_name
+										var/datum/dna/lottery = H.dna
+										lottery.transfer_identity(sinPerson, transfer_SE=1)
+										sinPerson.updateappearance(mutcolor_update=1)
+										sinPerson.domutcheck()
+						else
+							src.say("Your sin will be punished, [sinPerson]!")
+							var/sinPersonspecies = pick(species_list)
```
Comment (Shadowlight213) [inline code/game/gamemodes/miniantags/hades/hades.dm]: This needs to be limited to the bad species or something.  It's not much of a punishment if they get the agent species.

Comment (Iamgoofball) [inline code/game/gamemodes/miniantags/hades/hades.dm]: yeah but you could also die by bursting into flames as a plasmaman so i mean

Comment (Shadowlight213) [inline code/game/gamemodes/miniantags/hades/hades.dm]: Yeah, that's why I'm saying limit it to like I guess species that aren't blacklisted.

Comment (sawu-tg) [inline code/game/gamemodes/miniantags/hades/hades.dm]: To be Honest @Shadowlight213 in a large set of cases it'll be bad, so I don't think a small chance of it being not-so-bad would be terrible.

File: code/game/gamemodes/miniantags/hades/hades.dm (hunk 8 above)
Comment (optimumtact) [inline code/game/gamemodes/miniantags/hades/hades.dm]: use the NORTH/SOUTH/WEST/EAST defines surely

**Comment by Supermichael777** [comment]
but Hades was not the judge of the dead, merely the lord of the land. you want Minos, Rhadamanthys, and Aeacus

**Comment by sawu-tg** [comment]
@Supermichael777 Factual accuracy isn't my goal.

**Pull request merged**